src/core_scan/dune

```diff
@@ -4,6 +4,8 @@
  (wrapped false)
  (libraries
     parmap
+    trace
```

Comment:
we might want to define a libs/tracing/Tracing.ml that slightly wrap ocaml-trace, so we can customize things if we need, and this libs/tracing/dune can include trace and trace-def so then we only need to use 'tracing' instead of those 2 trace module in src/

Reply:
The advantage of wrapping ocaml-trace, is that we can change later to another underlying library
(e.g., jane street one).